### PR TITLE
Allow specifying path to already mounted BPF filesystem

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Comment by tgraf** [comment]
@aanm Are we waiting on something in particular to merge this change? This would unblock the DaemonSet patches.

**Comment by aanm** [comment]
> This would unblock the DaemonSet patches.

I thought so but the problem still persists even if this PR is merged or not.

File: cilium/Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@ clean:
 
 SOURCES := $(shell find monitor/ policy/ endpoint/ ../bpf ../daemon ../common ../pkg . -name '*.go')
 
+
```
Comment (aanm) [inline cilium/Makefile]: Unnecessary

File: pkg/bpf/bpffs.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+var (
+	// Path to where bpffs is mounted (default: /sys/fs/bpf)
+	mapRoot = "/sys/fs/bpf"
+
+	// Prefix for all maps (default: tc/globals)
+	mapPrefix = "tc/globals"
+
+	// Set to true on first get request to detect misorder
+	lockedDown = false
```
Comment (aanm) [inline pkg/bpf/bpffs.go:32]: Isn't `sync.Once` useful here?

Comment (tgraf) [reply pkg/bpf/bpffs.go:32]: How?

Comment (aanm) [reply pkg/bpf/bpffs.go:32]:
```golang
....
import "sync"

var (
        once sync.Once
)
....
func GetMapRoot() string {
        once.Do(func(){
         	lockedDown = true
         })
	return mapRoot
}
```

**Pull request merged**